### DISCUSSION TITLE
Request sha256 and ripemd160

### DISCUSSION THREAD

**Discussion by lity** [wevm/viem#1581] [Idea / Feature Request] [answered]
In Solidity, there are built-in functions for sha256 and ripemd160. There are also corresponding utility functions in the ethers library.

I have switched to Viem from Ethers, when I need sha256, where is it?

**Comment by jxom** [comment]
Use @noble/hashes: https://github.com/paulmillr/noble-hashes

**Reply by lity** [reply]
Yes, that's what I do now, but why not include them in viem?  keccak256, sha256, ripemd160, these basic hash functions are supposed to be in the same place.

**Reply by jxom** [reply]
We don't really want to extend the API for hashing functions that we don't use internally in Viem. Viem isn't a cryptography library and we just re-export `keccak256` from `@noble/hashes`.

**Reply by lity** [reply]
This makes sense. However, from the perspective of design consistency, either all of keccak256, sha256, and ripemd160 should be re-exported from @noble/hashes, or none of them should be re-exported. Only re-exporting keccak256 might seem a bit odd to users.

I believe that re-exporting all three hash functions, keccak256, sha256, and ripemd160, could be a better design choice because the Viem library interfaces with smart contracts, and keccak256, sha256, and ripemd160 are all built-in functions in Solidity.

**Reply by jxom** [reply]
Yeah, fair perspective. Don't think it would be too much of a hassle to extend the API with re-exports. Will look into it tomorrow.

**Reply by jxom** [reply, marked as answer]
#1589
